packages/material-ui/src/SwipeableDrawer/SwipeableDrawer.js

```diff
@@ -341,6 +341,17 @@ class SwipeableDrawer extends React.Component {
     } = this.props;
     const { maybeSwiping } = this.state;
 
+    let defaultPaperPropStyle = {
+      pointerEvents: variant === 'temporary' && !open ? 'none' : '',
+    };
+
+    if (PaperProps && PaperProps.style) {
```

Comment:
`PaperProps` will never be undefined, only the `PaperProps.style` check is needed here. Also, destructuring `null` or `undefined` is perfectly fine. This could be simplified (and inlined) to

```
style: { ...PaperProps.style, pointerEvents: variant === 'temporary' && !open ? 'none' : '' }
```

Reply:
I tested locally on the documentation and I had a `PaperProps` undefined that's but I'll double check that thanks 👍

Reply:
This should do it:

```diff
-      PaperProps,
+      PaperProps = {},
       swipeAreaWidth,
       variant,
       ...other
     } = this.props; 

// ...

           PaperProps={{
             ...PaperProps,
+            style: {
+              pointerEvents: variant === 'temporary' && !open ? 'none' : '',
+              ...PaperProps.style,
+            }
```